integration/yarn/src/main/java/alluxio/yarn/ContainerAllocator.java

```diff
@@ -139,7 +140,8 @@ public synchronized void allocateContainer(Container container) {
     mOutstandingContainerRequestsLatch.countDown();
   }
 
-  private void requestContainers(int numContainersToRequest) throws Exception {
+  private void requestContainers() throws Exception {
+    int numContainersToRequest = mTargetNumContainers - mAllocatedContainers.size();
```

Comment:
this needs to be set after calling `getPotentialWorkerHosts` to avoid the race condition where a new host is allocated after calculating `numContainersToRequest` but before calculating the unused hosts.